android/authentication.md

```diff
@@ -1,615 +1,704 @@
-        <intent-filter>
-            <action android:name="android.intent.action.MAIN" />
-            <category android:name="android.intent.category.LAUNCHER" />
-        </intent-filter>
-    </activity>
-    ```
-6. Update the `onCreate` function of your `AuthenticatorActivity` to call `AWSMobileClient`. This component provides the functionality to resume a signed-in authentication session. It makes a network call to retrieve the AWS credentials that allow users to access your AWS resources and registers a callback for when that transaction is complete.
-If the user is already signed in, the app switches to the `NextActivity`.  If the user isn't signed in, the user is presented with the AWS Mobile configurable sign-in UI.  After the user is authenticated, the app continues to the `NextActivity`.
+Note that the CLI allows you to select more than one identity provider for your app. You can also run `amplify auth update` to add an identity provider to an existing auth configuration.
 
+#### Google with Cognito Identity
 
-```java
-import android.app.Activity;
-import android.os.Bundle;
+To federate Google as a user sign-in provider for AWS services called in your app, you will pass tokens to `AWSMobileClient.getInstance().federatedSignIn()`. You must first register your application with Google Sign-In and then configure this with Amazon Cognito Identity Pools.
 
-import com.amazonaws.mobile.auth.ui.SignInUI;
-import com.amazonaws.mobile.client.AWSMobileClient;
-import com.amazonaws.mobile.client.AWSStartupHandler;
-import com.amazonaws.mobile.client.AWSStartupResult;
+The following sections detail the Google Sign-In requirements and steps to integrate Google Sign-In for both iOS and Android apps:
 
-public class AuthenticatorActivity extends Activity {
-    @Override
-    protected void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_authenticator);
+* [Create a Google Developers Project and OAuth Web Client ID](./auth-google-setup#create-a-google-developers-project-and-oauth-web-client-id) (required for `all apps` regardless of platform)
 
-        // Add a call to initialize AWSMobileClient
-        AWSMobileClient.getInstance().initialize(this, new AWSStartupHandler() {
-            @Override
-            public void onComplete(AWSStartupResult awsStartupResult) {
-                SignInUI signin = (SignInUI) AWSMobileClient.getInstance().getClient(
-                      AuthenticatorActivity.this,
-                      SignInUI.class);
-                signin.login(
-                      AuthenticatorActivity.this,
-                      NextActivity.class).execute();
-            }
-        }).execute();
-    }
-}
+* [Create an OAuth Android Client ID](./auth-google-setup#create-an-oauth-android-client-id) (required for all Android apps)
+
+* [Create an OAuth iOS Client ID](./auth-google-setup#create-an-oauth-ios-client-id) (required for all iOS apps)
```

Comment:
is this needed?

Reply:
Per-previous comment, I'm not sure. Let's talk.